extensions/hibernate-search-backend-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/backend/elasticsearch/runtime/HibernateSearchBackendElasticsearchRuntimeConfig.java

```diff
@@ -0,0 +1,351 @@
+    // otherwise "quarkus.hibernate-search-orm.elasticsearch.index-defaults" will be ignored.
+    @ConfigGroup
+    interface SchemaManagementConfig {
+        /**
+         * The minimal https://www.elastic.co/guide/en/elasticsearch/reference/7.17/cluster-health.html[Elasticsearch cluster
+         * status] required on startup.
+         *
+         * @asciidoclet
+         */
+        // We can't set an actual default value here: see comment on this class.
+        @ConfigDocDefault("yellow")
+        Optional<IndexStatus> requiredStatus();
+
+        /**
+         * How long we should wait for the status before failing the bootstrap.
+         */
+        // We can't set an actual default value here: see comment on this class.
+        @ConfigDocDefault("10S")
+        Optional<Duration> requiredStatusWaitTimeout();
+    }
+
+    // We can't set actual default values in this section,
+    // otherwise "quarkus.hibernate-search-orm.elasticsearch.index-defaults" will be ignored.
+    @ConfigGroup
+    interface IndexingConfig {
+        /**
+         * The number of indexing queues assigned to each index.
+         *
+         * Higher values will lead to more connections being used in parallel,
+         * which may lead to higher indexing throughput,
+         * but incurs a risk of overloading Elasticsearch,
+         * i.e. of overflowing its HTTP request buffers and tripping
+         * https://www.elastic.co/guide/en/elasticsearch/reference/7.9/circuit-breaker.html[circuit breakers],
+         * leading to Elasticsearch giving up on some request and resulting in indexing failures.
+         *
+         * @asciidoclet
+         */
+        // We can't set an actual default value here: see comment on this class.
+        @ConfigDocDefault("10")
+        OptionalInt queueCount();
+
+        /**
+         * The size of indexing queues.
+         *
+         * Lower values may lead to lower memory usage, especially if there are many queues,
+         * but values that are too low will reduce the likeliness of reaching the max bulk size
+         * and increase the likeliness of application threads blocking because the queue is full,
+         * which may lead to lower indexing throughput.
+         *
+         * @asciidoclet
+         */
+        // We can't set an actual default value here: see comment on this class.
+        @ConfigDocDefault("1000")
+        OptionalInt queueSize();
+
+        /**
+         * The maximum size of bulk requests created when processing indexing queues.
+         *
+         * Higher values will lead to more documents being sent in each HTTP request sent to Elasticsearch,
+         * which may lead to higher indexing throughput,
+         * but incurs a risk of overloading Elasticsearch,
+         * i.e. of overflowing its HTTP request buffers and tripping
+         * https://www.elastic.co/guide/en/elasticsearch/reference/7.9/circuit-breaker.html[circuit breakers],
```

Comment:
I wonder... (not for this PR)
since this is an asciidoc comment, should we do something similar as with `hibernate-search-docs-url` or maybe just point to `current` (https://www.elastic.co/guide/en/elasticsearch/reference/current/circuit-breaker.html)
(I thought about it as 7.9 is slightly EOLed 🙈 😃 )

Reply:
I agree it's a good idea, but would likely require moving Maven properties for the Elasticsearch version to the root POM: for a different PR indeed.